packages/client/hmi-client/src/workflow/tera-workflow.vue

```diff
@@ -429,29 +433,34 @@ function appendOutputPort(
 	// should be built into the Operation directly. What we are doing is to update the internal state
 	// and this feels it is leaking too much low-level information
 	if (
-		node.operationType === WorkflowOperationTypes.SIMULATE_JULIA ||
-		node.operationType === WorkflowOperationTypes.SIMULATE_CIEMSS ||
 		node.operationType === WorkflowOperationTypes.CALIBRATION_JULIA ||
 		node.operationType === WorkflowOperationTypes.CALIBRATION_CIEMSS
 	) {
-		const state = node.state as SimulateJuliaOperationState;
+		const state = node.state as CalibrationOperationStateJulia;
 		if (state.chartConfigs.length === 0) {
 			// This only ends up showing the output of the first run, perhaps we should consider showing
 			// the output of the last run, or all runs?
 			state.chartConfigs.push({
 				selectedRun: port.value[0],
 				selectedVariable: []
 			});
-		} else if (
-			node.operationType === WorkflowOperationTypes.SIMULATE_JULIA ||
-			node.operationType === WorkflowOperationTypes.SIMULATE_CIEMSS
-		) {
-			state.chartConfigs.push({
-				selectedRun: port.value[0],
-				selectedVariable: []
-			});
 		}
 	}
+
+	if (
+		node.operationType === WorkflowOperationTypes.SIMULATE_JULIA ||
+		node.operationType === WorkflowOperationTypes.SIMULATE_CIEMSS
+	) {
+		const state = node.state as SimulateJuliaOperationState;
+		if (state.simConfigs.chartConfigs.length === 0) {
+			state.simConfigs.chartConfigs.push([]);
+		}
+		state.simConfigs.runConfigs[port.value[0]] = {
+			runId: port.value[0],
+			active: true
+		};
+	}
+
```

Comment:
This section feels a little strange, as per the FIXME comment on the big if-block. Ideally we don't want to mess with state logic at this level, and now that state-logic is even more complex than before only exacerbates the situation. 

Let have a chat and see if we can work around this.

Reply:
moved this logic out into `tera-simulate-node-julia.vue` and `tera-simulate-node-ciemss.vue`